oras/provider.py

```diff
@@ -78,7 +78,7 @@ def __init__(
         self.session.cookies.set_policy(DefaultCookiePolicy(allowed_domains=[]))
 
         # Get custom backend, pass on session to share
-        self.auth = oras.auth.get_auth_backend(auth_backend, self.session)
+        self.auth = oras.auth.get_auth_backend(auth_backend, self.session, self.prefix)
```

Comment:
step 1.

the prefix of provider is set a few lines above:

oras.auth.get_auth_backend

and is "pass forward" as suggested to the call of `oras.auth.get_auth_backend()`